data/pools/transactionPool.go

```diff
@@ -295,35 +233,137 @@ func (pool *TransactionPool) Remember(txgroup []transactions.SignedTxn) error {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
 
-	err := pool.test(txgroup)
-	if err != nil {
-		return fmt.Errorf("TransactionPool.Remember: %v", err)
+	return pool.test(txgroup)
+}
+
+type poolIngestParams struct {
+	remember bool // if set, remember txn
+	checkFee bool // if set, perform fee checks
+	mustSync bool // if set, wait until ledger is caught up
+}
+
+// remember attempts to add a transaction group to the pool.
+func (pool *TransactionPool) remember(txgroup []transactions.SignedTxn) error {
+	params := poolIngestParams{
+		remember: true,
+		checkFee: true,
+		mustSync: true,
+	}
+	return pool.ingest(txgroup, params)
+}
+
+// add tries to add the transaction group to the pool, bypassing the fee
+// priority checks.
+func (pool *TransactionPool) add(txgroup []transactions.SignedTxn) error {
+	params := poolIngestParams{
+		remember: true,
+		checkFee: false,
+		mustSync: false,
 	}
+	return pool.ingest(txgroup, params)
+}
 
+// test checks whether a transaction group could be remembered in the pool,
+// but does not actually store this transaction in the pool.
+func (pool *TransactionPool) test(txgroup []transactions.SignedTxn) error {
+	params := poolIngestParams{
+		remember: false,
+		checkFee: true,
+		mustSync: true,
+	}
+	return pool.ingest(txgroup, params)
+}
+
+// ingest checks whether a transaction group could be remembered in the pool,
+// and stores this transaction in the pool if remember is set.
+//
+// ingest assumes that pool.mu is locked.  It might release the lock
+// while it waits for OnNewBlock() to be called.
+func (pool *TransactionPool) ingest(txgroup []transactions.SignedTxn, params poolIngestParams) error {
 	if pool.pendingBlockEvaluator == nil {
-		return fmt.Errorf("TransactionPool.Remember: no pending block evaluator")
+		return fmt.Errorf("TransactionPool.test: no pending block evaluator")
 	}
 
-	err = pool.remember(txgroup)
-	if err != nil {
-		return fmt.Errorf("TransactionPool.Remember: %v", err)
+	if params.mustSync {
```

Comment:
Since this code doesn't actually ensure sync (it just waits a bit), maybe it's better to rename this parameter to `preferSync`